src/github.rs

```diff
@@ -139,29 +142,58 @@ pub fn validate_pull_request(job: &worker::PullRequestJob, client: &Github) -> R
     Ok(failures)
 }
 
+pub trait TryExecute: Executor {
+    fn try_execute<T: DeserializeOwned>(self) -> Result<T>
+    where
+        Self: Sized
+    {
+        #[derive(Deserialize)]
+        struct GithubError {
+            message: String,
+        }
+
+        (|| match self.execute::<serde_json::Value>() {
+            Ok((_, StatusCode::Ok, Some(response))) => {
+                serde_json::from_value(response).chain_err(|| "Failed to parse response")
+            }
+            Ok((_, _, Some(response))) => {
+                Err(serde_json::from_value::<GithubError>(response)
+                    .chain_err(|| "Failed to parse error response")?
+                    .message
+                    .into())
+            }
+            Ok((_, _, None)) => Err("Received error response from github with no message".into()),
+            Err(err) => Err(err).chain_err(|| "Failed to execute request")?,
+        })().or_else(|err| {
+            error!("Failed to execute request: {}", err);
+            Err(err)
+        })
+    }
+}
+
+impl<'a> TryExecute for ::github_rs::repos::get::ContentsReference<'a> {}
+impl<'a> TryExecute for ::github_rs::repos::get::PullsNumber<'a> {}
+impl<'a> TryExecute for ::github_rs::repos::get::CollaboratorsUsernamePermission<'a> {}
+impl<'a> TryExecute for ::github_rs::repos::get::IssuesNumberComments<'a> {}
+impl<'a> TryExecute for ::github_rs::repos::get::PullsNumberCommits<'a> {}
```

Comment:
now that I'm looking at it, this makes sense. I don't think there would be another way to do it while preserving the chaining behavior you have below. It needs to be explicitly stated because you are defining a function where these types are the receivers, which is always explicitly stated in rust, even if it ends up just being the default implementation of the function.